### PR TITLE
Fix etcd failures in ci-kubernetes-e2e-cos-gce-disruptive-canary

### PR DESCRIPTION
**Pull request by dims**
etcd_failure.go was assuming that the API address for the cluster is where etcd process runs which is not true. So just like we did in daemon_restart.go, we should find the external IP for the control plane nodes and use that one.

- Reuse the same bits of code across etcd_failure.go, daemon_restart.go and controlplane_nodes.test.go
- Move the shared code into a common place.
- Switch etcd_failure.go to use the new methods for the external ip for the control plane node.
- Run this test on `aws` as well!

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -128,6 +129,8 @@ const (
 
 	// SnapshotDeleteTimeout is how long for snapshot to delete snapshotContent.
 	SnapshotDeleteTimeout = 5 * time.Minute
+
+	ControlPlaneLabel = "node-role.kubernetes.io/control-plane"
```
Comment (neolit123) [inline test/e2e/framework/util.go:133]: IIRC, this label is a kubeadm-ism and we cannot assume all k8s deployers apply it
https://github.com/kubernetes/enhancements/blob/62cad7da88a76ae416e749311a59a91ec4a0bab5/keps/sig-architecture/1143-node-role-labels/README.md?plain=1

if this is for our internal testing with kubeadm, kops we can assume it. but not for the wider conformance.

Comment (dims) [reply test/e2e/framework/util.go:133]: @neolit123 ack, will add a explicit comment that this is for `kops` and `kubeadm` based clusters.

**Comment by dims** [comment]
both the etcd tests passed!

image

**Comment by dims** [comment]
/hold

to incorporate comment from https://github.com/kubernetes/kubernetes/pull/127033#issuecomment-2324686960

**Review by upodroid — APPROVED** [review]
Thank you for working on the kubeup to kops migration.

/lgtm

**Comment by dims** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/127055/pull-kubernetes-e2e-gce-disruptive-canary/1830598265465212928 looks good

(just the already currently failing `sig-cloud-provider-gcp` and currently flaky `should assign default SC to PVCs that have no SC set` are failing)

**Pull request merged**